src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (typeof compareFunction === 'function') {
+      for (let i = 0; i < this.length; i++) {
+        for (let j = i + 1; j < this.length; j++) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const temp = this[j];
+
+            this[j] = this[i];
+            this[i] = temp;
+          }
+        }
+      }
+
+      return this;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (this[i].toString() > this[j].toString()) {
```

Comment:
Using 'toString()' for comparison will work for primitive types, but it may not work as expected for objects. Consider the case where objects need to be sorted based on one of their properties.